plugins/sam3dj.js

```diff
@@ -1,353 +1,401 @@
+		icon: 'print',
+		onload() {
+			MenuBar.addAction(btnExport, 'file.export');
+		},
+		onunload() {
+			MenuBar.removeAction('file.export.export_3dj');
+		}
+	};
+	const SC_TEXTURE_BLANK = 'sc-peripherals:block/white';
+
+	/**
+	 * Returns a fitting texture for a given cube.
+	 * @param {*} cube The cube, which holds the wanted texture.
+	 * @returns The wanted texture, fitting to the cube. (+ some booleans) in a object.
+	 */
+	function getSingleTexture(cube, scan) {
+		let faces = 0;
+		let missing = false;
+		let texture = SC_TEXTURE_BLANK;
+
+		for (let dir in cube.faces) { //in ['north', 'east', 'south', 'west', 'up', 'down']
+			let tmp = cube.faces[dir].getTexture();
+			if (tmp === undefined) {
+				missing = true;
+				continue;
+			}
+			let logOutput = ""
+			// texture: <namespace>:<path>
+			let namespace = ""
+			let id = ""
+			if (tmp.name.includes(":")) {
+				id = tmp.name
+			} else if (tmp.name.includes("/")) {
+				id = "minecraft:" + tmp.name
+			}
+			if (id === "") {
+				if (cube.name.includes(":")) {
+					id = cube.name.replace(/\.[^/.]+$/, "")
+					logOutput = "Exporting " + cube.name + " as " + id
+				} else {
+					if (typeof cube.parent === 'object') {
+						let parents = []
+						let parent = cube.parent
+						while (typeof parent === 'object') {
+							parents.push(parent.name)
+							parent = parent.parent
+						}
+						parents.reverse()
+						namespace = parents[0]
+						parents.shift()
+						path = parents.join("/")
+						if (path !== "") {
+							path += "/"
+						}
+						id = namespace + ":" + path + cube.name
+						logOutput = "Exporting " + cube.name + " in " + namespace + "/" + path + " as " + id
+					} else {
+						if (cube.name.includes(":")) {
+							id = cube.name
+							logOutput = "Exporting " + cube.name + " as " + id
+						} else if (cube.name.includes("/")) {
+							id = "minecraft:" + cube.name
+							logOutput = "Exporting " + cube.name + " as " + id
+						} else {
+							id = "minecraft:block/" + cube.name
+							logOutput = "Exporting " + cube.name + " as " + id
+						}
+					}
+				}
+			} else {
+				logOutput = "Exporting " + cube.name + " as " + id
+			}
+
+			if (scan && logOutput !== "") {
+				console.log(logOutput)
+			}
+
+			if (texture !== id) {
+				faces++;
+				texture = id;
+			}
+		}
+
+
+		return {
+			texture: texture,
+			missingTexture: missing,
+			multipleTexture: (faces > 1),
+		};
+	}
+
+
+	/**
+	 * Converts a given element to a shape in .3DJ format.
+	 * @param {*} obj The given element (e.g. Cube).
+	 * @returns The fianl shape (+ some booleans) in a object.
+	 */
+	function genShape(cube, raw = false) {
+		let result = getSingleTexture(cube);
+		let shape = {
+			bounds: [
```

Comment:
Like I said in my comment, the offset depends on the model format, this should be accounted for, as otherwise users may run into issues.